tensorflow-queue/tests/QueueTest.hs

```diff
@@ -39,42 +41,45 @@ import qualified Data.ByteString as BS
 -- | Test basic queue behaviors.
 testBasic :: Test
 testBasic = testCase "testBasic" $ runSession $ do
-    (q :: Queue2 Int64 BS.ByteString) <- build $ makeQueue2 1 ""
-    buildAnd run_ (enqueue q 42 (scalar "Hi"))
+    q :: Queue [Int64, BS.ByteString] <- build $ makeQueue 1 ""
+    buildAnd run_ $ enqueue q $ 42 :| scalar "Hi" :| Nil
     x <- buildAnd run (dequeue q)
-    liftIO $ (Scalar 42, Scalar "Hi") @=? x
+    liftIO $ (Identity (Scalar 42) :| Identity (Scalar "Hi") :| Nil) @=? x
 
-    buildAnd run_ (enqueue q 56 (scalar "Bar"))
+    buildAnd run_ $ enqueue q $ 56 :| scalar "Bar" :| Nil
     y <- buildAnd run (dequeue q)
-    liftIO $ (Scalar 56, Scalar "Bar") @=? y
+    let expected = Identity (Scalar 56) :| Identity (Scalar "Bar") :| Nil
+    liftIO $ expected @=? y
 
 -- | Test queue pumping.
 testPump :: Test
 testPump = testCase "testPump" $ runSession $ do
     (deq, pump) <- build $ do
-        q :: Queue2 Int64 BS.ByteString <- makeQueue2 2 "ThePumpQueue"
+        q :: Queue [Int64, BS.ByteString] <- makeQueue 2 "ThePumpQueue"
         (,) <$> dequeue q
-            <*> enqueue q 31 (scalar "Baz")
+            <*> enqueue q (31 :| scalar "Baz" :| Nil)
     -- This is a realistic use. The pump inputs are pre-bound to some
     -- nodes that produce values when pumped (e.g. read from a
     -- file).
     run_ (pump, pump)
 
     (x, y) <- run (deq, deq)
-    liftIO $ (Scalar 31, Scalar "Baz") @=? x
-    liftIO $ (Scalar 31, Scalar "Baz") @=? y
+    let expected = Identity (Scalar 31) :| Identity (Scalar "Baz") :| Nil
+    liftIO $ expected @=? x
+    liftIO $ expected @=? y
 
 testAsync :: Test
 testAsync = testCase "testAsync" $ runSession $ do
     (deq, pump) <- build $ do
-        q :: Queue2 Int64 BS.ByteString <- makeQueue2 2 ""
+        q :: Queue [Int64, BS.ByteString] <- makeQueue 2 ""
         (,) <$> dequeue q
-            <*> enqueue q 10 (scalar "Async")
+            <*> enqueue q (10 :| scalar "Async" :| Nil)
```

Comment:
It would be really good to explain why this 10 doesn't need any Identity whereas Scalar 10 below does.